packages/cozy-realtime/src/CozyRealtime.js

```diff
@@ -1,360 +1,534 @@
 
-      this._socket.once(`subscribe_${type}_${id}`, resolve)
-      this._socket.subscribe(type, id)
-    })
+  /**
+   * Register event handlers for cozy-client events
+   * @private
+   */
+  subscribeClientEvents() {
+    this.client.on('login', this.onClientLogin)
+    this.client.on('tokenRefreshed', this.onClientTokenRefreshed)
+    this.client.on('logout', this.onClientLogout)
   }
 
   /**
-   * Remove the given handler from the list of handlers for given
-   * doctype/document and event.
-   *
-   * @param {String}  type      Document doctype to unsubscribe from
-   * @param {String}  id        Document id to unsubscribe from
-   * @param {String}  eventName Event to unsubscribe from
-   * @param {Function}  handler   Function to call when an event of the
-   * given type on the given doctype or document is received from stack.
+   * Unregister event handlers for cozy-client events
+   * @private
    */
-  unsubscribe(eventName, type, handlerOrId, handlerOrUndefined) {
-    const { handler, id } = getHandlerAndId(handlerOrId, handlerOrUndefined)
-    validateParameters(eventName, type, id, handler)
-    const key = generateKey(eventName, type, id)
-
-    this.removeListener(key, handler)
+  unsubscribeClientEvents() {
+    this.client.removeListener('login', this.onClientLogin)
+    this.client.removeListener('tokenRefreshed', this.onClientTokenRefreshed)
+    this.client.removeListener('logout', this.onClientLogout)
+  }
 
-    if (!this._haveEventHandler()) {
-      this._resetSocket()
+  /**
+   * When the cozy-client instance successfully login - re-authenticate
+   * @private
+   */
+  onClientLogin() {
+    logger.info('Received login from cozy-client')
+    if (this.isWebSocketOpen()) {
+      this.authenticate()
+      // send subscriptions again, permissions may have changed
+      this.sendSubscriptions()
     }
   }
 
   /**
-   * Unsubscribe all handlers and close socket
+   * When the cozy-client instance refresh its access token - re-authenticate
+   * @private
    */
-  unsubscribeAll() {
-    this._getEventKeys()
-      .map(key => this._events[key].length > 0 && key)
-      .filter(Boolean)
-      .forEach(key => (this._events[key] = []))
+  onClientTokenRefreshed() {
+    // no need to authenticate again
+  }
 
-    this._resetSocket()
+  /**
+   * When the cozy-client instance logs out - throw the socket, don't reconnect yet (no authentication)
+   * @private
+   */
+  onClientLogout() {
+    if (this.hasWebSocket()) {
+      this.rejectWebSocket()
+    }
   }
 
-  async send(doctype, id, data) {
-    return this._cozyClient.stackClient.fetchJSON(
-      'POST',
-      `/realtime/${doctype}/${id}`,
-      {
-        data
-      }
-    )
+  /* * * * * * * * * * * * * * * */
+  /* * Global (browser) EVENTS * */
+  /* * * * * * * * * * * * * * * */
+
+  /**
+   * Subscribe to global events
+   * @private
+   */
+  subscribeGlobalEvents() {
+    if (isInBrowser && window.addEventListener) {
```

Comment:
I'm not confortable with the fact that we only handle the browser use case. Realtime should manage our mobile app correctly

Reply:
Wo, ok. isBrowser only test for window !== undefined. I don't like the naming then.

Reply:
would you be more confortable with `hasBrowserContext` ?

Reply:
Yep 👍

Reply:
done